src/network/netdev/batadv.c

```diff
@@ -0,0 +1,205 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <inttypes.h>
+
+#include "netdev/batadv.h"
+#include "network-internal.h"
+#include "networkd-manager.h"
+#include "netlink-util.h"
+#include <linux/genetlink.h>
+
+#include "fileio.h"
+#include "parse-util.h"
+#include "stdio-util.h"
+#include "string-util.h"
+
+static void batadv_init(NetDev *n) {
+        _unused_ Batadv *b;
+
+        b = BATADV(n);
+
+        /* Set defaults */
+        b->aggregation=true;
+        b->bridge_loop_avoidance=true;
+        b->distributed_arp_table=true;
+        b->fragmentation=true;
+        b->hop_penalty=true;
+}
+
+int config_parse_batadv_gateway_mode(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Batadv *b = userdata;
+        int mode;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        mode = BATADV_GW_MODE_OFF;
+
+        if (streq(rvalue, "client"))
+                mode = BATADV_GW_MODE_CLIENT;
+
+        else if (streq(rvalue, "server"))
+                mode = BATADV_GW_MODE_SERVER;
+
+        else {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                          "Batadv gateway mode '%s' unknown. Ignoring.", rvalue);
+                return 0;
+        }
+
+        b->gateway_mode = mode;
+
+        return 0;
+}
+
+int config_parse_badadv_bandwidth (
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint64_t k;
+        uint64_t bandwidth;
```

Comment:
hmm, 32bit or 64? the fields in the Batadv structure are 32bit, but here you use 64bit? what is it?

Reply:
Just looked it up it seems batadv only accepts 32bit here so fixed it.